Jenkinsfile

```diff
@@ -0,0 +1,106 @@
+//"Jenkins Pipeline is a suite of plugins which supports implementing and integrating continuous delivery pipelines into Jenkins. Pipeline provides an extensible set of tools for modeling delivery pipelines "as code" via the Pipeline DSL."
+//More information can be found on the Jenkins Documentation page https://jenkins.io/doc/
+
+@Library('github.com/connexta/cx-pipeline-library@master') _
+@Library('github.com/connexta/github-utils-shared-library@master') __
+
+pipeline {
+    agent {
+        node {
+            label 'linux-large-ddf'
+            customWorkspace "/jenkins/workspace/${JOB_NAME}/${BUILD_NUMBER}"
+        }
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'25'))
+        disableConcurrentBuilds()
+        timestamps()
+        skipDefaultCheckout()
+    }
+    triggers {
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(BRANCH_NAME == "master" ? "H H(17-19) * * *" : "")
+    }
+    environment {
+        LARGE_MVN_OPTS = '-Xmx4G -Xms1G -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        DISABLE_DOWNLOAD_PROGRESS_OPTS = '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        COVERAGE_EXCLUSIONS = '**/test/**/*,**/itests/**/*,**/*Test*,**/sdk/**/*,**/*.js,**/node_modules/**/*,**/jaxb/**/*,**/wsdl/**/*,**/nces/sws/**/*,**/*.adoc,**/*.txt,**/*.xml'
+        GITHUB_USERNAME = 'codice'
+        GITHUB_TOKEN = credentials('cxddfuibot')
+        GITHUB_REPONAME = 'ddf-admin-ui'
+    }
+    stages {
+        stage('Setup') {
+            steps {
+                dockerd {}
+                slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
+                postCommentIfPR("Internal build has been started, your results will be available at build completion.", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+            }
+        }
+
+        // Checkout the repository
+        stage('Checkout repo') {
```

Comment:
Not sure we need this stage for a new repository. 
If you remove the `disableDefaultCheckout()` from the options block the declarative pipeline will checkout the repository for you.

Reply:
Good point. Do we need the retry logic? Do you know if we get that with the declarative pipeline?

Reply:
I dont know if "retry" is automatic by default in the declarative pipeline. With a smaller repo do you suspect that we will timeout in checking out code?

Reply:
I would hope we would not need it on the small repo. I'll try removing and add it back later if we need it.

Reply:
I removed this stage and the skipDefaultCheckout() line

Reply:
Looks good to me!
The logic to checkout the repo can be modified to ensure that only a single attempt is needed to clone.